Release v0.39.0 (#7987)
radius-project/radius: docs/release-notes/v0.39.0.md

# Announcing Radius v0.39.0

Today we're happy to announce the release of Radius v0.39.0. Check out the [highlights](#highlights) below, along with the [full changelog](#full-changelog) for more details.

We would like to extend our thanks to all the [new](#new-contributors) and existing contributors who helped make this release possible!

## Intro to Radius

If you're new to Radius, check out our website, [radapp.io](https://radapp.io), for more information. Also visit our [getting started guide](https://docs.radapp.io/getting-started/) to learn how to install Radius and create your first app.

## Highlights

### Support for Dapr Configuration Building Block

> A new `Applications.Dapr/configurationStores` resource type is now available, which enables built-in support for the Dapr [Configuration](https://docs.dapr.io/getting-started/quickstarts/configuration-quickstart/) building block in Radius. The Dapr Configuration building block allows for dynamic configuration updates, including feature flag management, and is a great addition to the set of supported Dapr features in Radius. To learn more, go to the [Dapr Configuration Store resource schema](https://docs.radapp.io/reference/resource-schema/dapr-schema/configurationstore/) in the Radius documentation.

Reviewer: ```suggestion
A new `Applications.Dapr/configurationStores` resource type is now available, which adds built-in support in Radius for the [Dapr Configuration](https://docs.dapr.io/getting-started/quickstarts/configuration-quickstart/) building block in Radius. The Dapr Configuration building block allows for dynamic configuration updates, including feature flag management, and is a great addition to the set of supported Dapr features in Radius. To learn more, go to the [Dapr Configuration Store resource schema](https://docs.radapp.io/reference/resource-schema/dapr-schema/configurationstore/) in the Radius documentation.
```

> A new `Applications.Dapr/configurationStores` resource type is now available, which adds built-in support in Radius for the [Dapr Configuration](https://docs.dapr.io/getting-started/quickstarts/configuration-quickstart/) building block in Radius. The Dapr Configuration building block allows for dynamic configuration updates, including feature flag management, and is a great addition to the set of supported Dapr features in Radius. To learn more, go to the [Dapr Configuration Store resource schema](https://docs.radapp.io/reference/resource-schema/dapr-schema/configurationstore/) in the Radius documentation.

Reviewer 2: ```suggestion
A new `Applications.Dapr/configurationStores` resource type is now available, to define and deploy [Dapr Configuration](https://docs.dapr.io/getting-started/quickstarts/configuration-quickstart/) building block in Radius. The Dapr Configuration building block allows for dynamic configuration updates, including feature flag management, and is a great addition to Dapr integration in Radius. To learn more, go to the [Dapr Configuration Store resource schema](https://docs.radapp.io/reference/resource-schema/dapr-schema/configurationstore/) in the Radius documentation. Thanks to community contributor @SoTrx for adding this functionality.
```

> A new `Applications.Dapr/configurationStores` resource type is now available, to define and deploy [Dapr Configuration](https://docs.dapr.io/getting-started/quickstarts/configuration-quickstart/) building block in Radius. The Dapr Configuration building block allows for dynamic configuration updates, including feature flag management, and is a great addition to the Dapr integration in Radius. To learn more, go to the [Dapr Configuration Store resource schema](https://docs.radapp.io/reference/resource-schema/dapr-schema/configurationstore/) in the Radius documentation. Thanks to community contributor @SoTrx for adding this functionality.

Reviewer: ```suggestion
A new `Applications.Dapr/configurationStores` resource type is now available to define and deploy the [Dapr Configuration](https://docs.dapr.io/getting-started/quickstarts/configuration-quickstart/) building block in Radius. The Dapr Configuration building block allows for dynamic configuration updates, including feature flag management, and is a great addition to the Dapr integration in Radius. To learn more, go to the [Dapr Configuration Store resource schema](https://docs.radapp.io/reference/resource-schema/dapr-schema/configurationstore/) in the Radius documentation. Thanks to community contributor @SoTrx for adding this functionality.
```

### In-memory data store for testing and development

> Radius has added an implementation for an in-memory version of the data store interface, which may be used for testing and development purposes. This helps to speed up and simplify the existing ETC.d and mocking mechanisms that are frequently used today in tests.

Reviewer 3: We should make it really clear that this is for contributors not users.

Author: `Radius has added an implementation for an in-memory version of the data store interface, which may be used for testing and development purposes, when contributing to Radius.`

Does this addition at the end make it clear @rynowak? @willtsai please update if you have thoughts on this.

Reviewer 3: I think that's good enough 👍

> Radius has added an implementation for an in-memory version of the data store interface, which may be used for testing and development purposes. This helps to speed up and simplify the existing ETC.d and mocking mechanisms that are frequently used today in tests.

Author: ```suggestion
Radius has added an implementation for an in-memory version of the data store interface, which may be used for testing and development purposes, when contributing to Radius. This helps to speed up and simplify the existing ETC.d and mocking mechanisms that are frequently used today in tests.
```

# Announcing Radius v0.39.0

Today we're happy to announce the release of Radius v0.39.0. Check out the [highlights](#highlights) below, along with the [full changelog](#full-changelog) for more details.

We would like to extend our thanks to all the [new](#new-contributors) and existing contributors who helped make this release possible!

## Intro to Radius

If you're new to Radius, check out our website, [radapp.io](https://radapp.io), for more information. Also visit our [getting started guide](https://docs.radapp.io/getting-started/) to learn how to install Radius and create your first app.

## Highlights

### Dapr Building Block: Configuration Store
https://github.com/radius-project/radius/pull/7906

### Add an in-memory data store
https://github.com/radius-project/radius/pull/7950

### Implement PostgreSQL datastore
https://github.com/radius-project/radius/pull/7961

## Breaking changes

> https://github.com/radius-project/radius/pull/7906

Author: @willtsai could you please help updating these. I have copied over PRs of a few items from changelog that might be good candidates for highlights.

> ## Highlights
> 
> ### Dapr Building Block: Configuration Store
> https://github.com/radius-project/radius/pull/7906
> 
> ### Add an in-memory data store
> https://github.com/radius-project/radius/pull/7950
> 
> ### Implement PostgreSQL datastore
> https://github.com/radius-project/radius/pull/7961
> 
> ## Breaking changes
> https://github.com/radius-project/radius/pull/7906

Reviewer: I think `Implement PostgreSQL datastore` is too early of an addition to call out in highlights, we should call it out when it's available for users to switch to a postgres datastore.

I don't think this is a breaking change, so I removed it: https://github.com/radius-project/radius/pull/7906

```suggestion
## Highlights

### Support for Dapr Configuration Building Block
A new `Applications.Dapr/configurationStores` resource type is now available, which enables built-in support for the Dapr [Configuration](https://docs.dapr.io/getting-started/quickstarts/configuration-quickstart/) building block in Radius. The Dapr Configuration building block allows for dynamic configuration updates, including feature flag management, and is a great addition to the set of supported Dapr features in Radius. To learn more, go to the [Dapr Configuration Store resource schema](https://docs.radapp.io/reference/resource-schema/dapr-schema/configurationstore/) in the Radius documentation.

### In-memory data store for testing and development
Radius has added an implementation for an in-memory version of the data store interface, which may be used for testing and development purposes. This helps to speed up and simplify the existing ETC.d and mocking mechanisms that are frequently used today in tests.

## Breaking changes
None.
```
